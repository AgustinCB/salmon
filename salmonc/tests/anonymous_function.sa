(fun(a) {
    print "hola";
    print a;
})(2);
print (fun(a) {
    print "hola";
    return a;
})(2);
