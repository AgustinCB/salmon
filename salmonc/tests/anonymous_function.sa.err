Constants: [Integer(2), Function { ip: 19, arity: 1 }, Integer(1), Integer(3), String(0), String(1), Function { ip: 44, arity: 1 }, String(5)]
Instructions: [Instruction { instruction_type: Constant(0), location: 0 }, Instruction { instruction_type: Constant(1), location: 0 }, Instruction { instruction_type: Call, location: 0 }, Instruction { instruction_type: Pop, location: 0 }, Instruction { instruction_type: Constant(4), location: 1 }, Instruction { instruction_type: Constant(0), location: 1 }, Instruction { instruction_type: Constant(6), location: 1 }, Instruction { instruction_type: Call, location: 1 }, Instruction { instruction_type: ToStr, location: 1 }, Instruction { instruction_type: StringConcat, location: 1 }, Instruction { instruction_type: Push, location: 1 }, Instruction { instruction_type: Strlen, location: 1 }, Instruction { instruction_type: Swap, location: 1 }, Instruction { instruction_type: Constant(2), location: 1 }, Instruction { instruction_type: Constant(3), location: 1 }, Instruction { instruction_type: Constant(2), location: 1 }, Instruction { instruction_type: Syscall, location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: Return, location: 1 }, Instruction { instruction_type: Constant(4), location: 0 }, Instruction { instruction_type: Constant(5), location: 0 }, Instruction { instruction_type: ToStr, location: 0 }, Instruction { instruction_type: StringConcat, location: 0 }, Instruction { instruction_type: Push, location: 0 }, Instruction { instruction_type: Strlen, location: 0 }, Instruction { instruction_type: Swap, location: 0 }, Instruction { instruction_type: Constant(2), location: 0 }, Instruction { instruction_type: Constant(3), location: 0 }, Instruction { instruction_type: Constant(2), location: 0 }, Instruction { instruction_type: Syscall, location: 0 }, Instruction { instruction_type: Pop, location: 0 }, Instruction { instruction_type: Constant(4), location: 0 }, Instruction { instruction_type: GetLocal(0), location: 0 }, Instruction { instruction_type: ToStr, location: 0 }, Instruction { instruction_type: StringConcat, location: 0 }, Instruction { instruction_type: Push, location: 0 }, Instruction { instruction_type: Strlen, location: 0 }, Instruction { instruction_type: Swap, location: 0 }, Instruction { instruction_type: Constant(2), location: 0 }, Instruction { instruction_type: Constant(3), location: 0 }, Instruction { instruction_type: Constant(2), location: 0 }, Instruction { instruction_type: Syscall, location: 0 }, Instruction { instruction_type: Pop, location: 0 }, Instruction { instruction_type: Return, location: 0 }, Instruction { instruction_type: Constant(4), location: 1 }, Instruction { instruction_type: Constant(5), location: 1 }, Instruction { instruction_type: ToStr, location: 1 }, Instruction { instruction_type: StringConcat, location: 1 }, Instruction { instruction_type: Push, location: 1 }, Instruction { instruction_type: Strlen, location: 1 }, Instruction { instruction_type: Swap, location: 1 }, Instruction { instruction_type: Constant(2), location: 1 }, Instruction { instruction_type: Constant(3), location: 1 }, Instruction { instruction_type: Constant(2), location: 1 }, Instruction { instruction_type: Syscall, location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: GetLocal(0), location: 1 }, Instruction { instruction_type: Return, location: 1 }]
Locations: [Location { address: 5, line: 0 }, Location { address: 5, line: 4 }]
