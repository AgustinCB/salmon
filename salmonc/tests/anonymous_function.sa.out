hola
2
hola
2
